File: src/main/java/infinitespire/relics/CubicDiamond.java
package infinitespire.relics;

import com.megacrit.cardcrawl.dungeons.AbstractDungeon;
import com.megacrit.cardcrawl.helpers.RelicLibrary;
import com.megacrit.cardcrawl.relics.AbstractRelic;

public class CubicDiamond extends Relic {

	public static final String ID = "Cubic Diamond";
	public static final String NAME = "Cubic Diamond";
	
	public CubicDiamond() {
		super(ID, "cubicdiamond", RelicTier.SPECIAL, LandingSound.CLINK);
	}

	@Override
	public AbstractRelic makeCopy() {
		return new CubicDiamond();
	}
	
	private AbstractRelic selectRelic(AbstractRelic original) {
		AbstractRelic relicToAdd = RelicLibrary.starterList.get(AbstractDungeon.cardRandomRng.random(RelicLibrary.starterList.size() - 1));;
		
		if(relicToAdd.relicId.equals(original.relicId)) {
			relicToAdd = selectRelic(original);
		}
		
		return relicToAdd;
	}

	//fix this to make it work right
	@Override
	public void onEquip() {
		AbstractRelic relicToAdd;
		for(AbstractRelic relic : AbstractDungeon.player.relics) {
			if(relic.tier == RelicTier.STARTER) {
				relicToAdd = selectRelic(relic);
				relicToAdd.instantObtain(AbstractDungeon.player, 0, false);
				relicToAdd.playLandingSFX();
				relicToAdd.flash();
				break;
			}
		}
	}
}

File: src/main/kotlin/com/blanktheevil/infinitespire/relics/CubicDiamond.kt
package com.blanktheevil.infinitespire.relics

import com.blanktheevil.infinitespire.extensions.getRandomItem
import com.blanktheevil.infinitespire.extensions.makeID
import com.blanktheevil.infinitespire.extensions.player
import com.megacrit.cardcrawl.dungeons.AbstractDungeon
import com.megacrit.cardcrawl.helpers.RelicLibrary
import com.megacrit.cardcrawl.relics.AbstractRelic
import com.megacrit.cardcrawl.relics.Circlet

class CubicDiamond : Relic(ID, IMG, TIER, SOUND){
  companion object {
    val ID = "Cubic Diamond".makeID()
    private const val IMG = "cubicdiamond"
    private val TIER = RelicTier.SPECIAL
    private val SOUND = LandingSound.CLINK
  }

  override fun onEquip() {
    val relicToReplaceIndex = player.relics.indexOfFirst { it.tier == RelicTier.STARTER }
    player.relics[relicToReplaceIndex].onUnequip()
    getUniqueStarterRelic(player.relics[relicToReplaceIndex]).apply {
      instantObtain(player, 0, true)
      playLandingSFX()
      flash()
    }
  }

  private fun getUniqueStarterRelic(original: AbstractRelic): AbstractRelic {
    val relicToAdd = RelicLibrary.starterList.getRandomItem(AbstractDungeon.relicRng) ?: Circlet()
    return if (relicToAdd.relicId == original.relicId) {
      getUniqueStarterRelic(original)
    } else {
      relicToAdd
    }
  }
}